src/controllers/ReviewPenaltyController.ts

```diff
@@ -23,22 +27,44 @@ const navigation = {
 export class ReviewPenaltyController extends SafeNavigationBaseController<PenaltyDetailsTable> {
 
     public static PENALTY_EXPECTED_ERROR: string = 'Penalty object expected but none found';
+    public static PENALTY_IDENTIFIER_EXPECTED_ERROR: string = 'User input penalty identifier expected';
+    public static PENALTY_NOT_FOUND: string = 'Penalty identifier did not match any item in list of penalties expected';
 
     constructor() {
         super(template, navigation);
     }
 
     public prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & PenaltyDetailsTable {
 
-        if (!appeal.penaltyIdentifier.penaltyList) {
+        const penaltyList = appeal.penaltyIdentifier.penaltyList;
+
+        if (!penaltyList) {
             throw new Error(ReviewPenaltyController.PENALTY_EXPECTED_ERROR);
+        }
+
+        let penaltyReference: string | undefined;
+
+        if (penaltyList.items.length === 1) {
+            penaltyReference = appeal.penaltyIdentifier.userInputPenaltyReference;
```

Comment:
if there is only one item in the penaltyList, at which point does the `appeal.penaltyIdentifier.penaltyReference` get set?

Reply:
In the processor of the SelectPenaltyController

Reply:
what happens if the `PenaltyReferenceRouter` has redirected us to this page from the penalty details page?

Reply:
then it must mean that there is only one penalty reference and it will show you the table as it should? What are you asking?

Reply:
i was just double checking that the `penaltyReference` is still stored in the session, as before